contracts/src/upgrades/Gateway202410.sol

```diff
@@ -25,10 +25,27 @@ contract Gateway202410 is Gateway {
     {}
 
     // Override parent initializer to prevent re-initialization of storage.
-    function initialize(bytes memory) external view override {
+    function initialize(bytes memory) external override {
         // Ensure that arbitrary users cannot initialize storage in this logic contract.
         if (ERC1967.load() == address(0)) {
             revert Unauthorized();
         }
+
+        // We expect version 0, deploying version 1.
+        CoreStorage.Layout storage $ = CoreStorage.layout();
```

Comment:
What's this for?

For the next upgrade after this, we'll deploy a new contract like `Gateway202506.sol` with different initializer code.

Reply:
It stops us from deploying this instance of Gateway202410 twice and hence running the same migration twice. I know this is controlled by governance, but an explicit way to do this in code is better.

Reply:
Sure, we can consider adding a version check, but I don't think this PR is the right place for it.

I will remove the version check for the time being.